beeline/src/java/org/apache/hive/beeline/JSONOutputFormat.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * This source file is based on code taken from SQLLine 1.9
+ * See SQLLine notice in LICENSE
+ */
+package org.apache.hive.beeline;
+
+import java.sql.SQLException;
+import java.sql.Types;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+
+import com.fasterxml.jackson.core.JsonEncoding;
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.JsonGenerator;
+
+/**
+ * OutputFormat for standard JSON format.
+ * 
+ */ 
+public class JSONOutputFormat extends AbstractOutputFormat {
+  protected final BeeLine beeLine;
+  protected JsonGenerator generator;
+  
+
+  /**
+   * @param beeLine
+   */
+  JSONOutputFormat(BeeLine beeLine){ 
+    this.beeLine = beeLine;
+    ByteArrayOutputStream buf = new ByteArrayOutputStream();
+    try {
+      this.generator = new JsonFactory().createGenerator(buf, JsonEncoding.UTF8);
+    } catch(IOException e) {
+      beeLine.handleException(e);
+    }
+  }
+
+  @Override 
+  void printHeader(Rows.Row header) {
+    try {
+      generator.writeStartObject();
+      generator.writeArrayFieldStart("resultset");
+    } catch(IOException e) {
+      beeLine.handleException(e);
+    }
+  }
+
+  @Override 
+  void printFooter(Rows.Row header) {
+    try {
+      generator.writeEndArray();
+      generator.writeEndObject();
+      generator.flush();
+      String out = ((ByteArrayOutputStream) generator.getOutputTarget()).toString("UTF-8");
```

Comment:
Check out `StandardCharsets.UTF_8`.  It's not a requirement that all JVMs support UTF-8.

https://docs.oracle.com/javase/7/docs/api/java/nio/charset/StandardCharsets.html

Reply:
The Javadocs for `ByteArrayOutputStream.toString(String charsetName)`[1] point to [this charset page](https://docs.oracle.com/javase/7/docs/api/java/nio/charset/Charset.html).
Which says that "UTF-8" as the correct canonical name for the StandardCharset. I _think_ my implementation is correct for all JVM's?

[1] https://docs.oracle.com/javase/7/docs/api/java/io/ByteArrayOutputStream.html

Reply:
For clarities sake I'm changing this line to `String out = ((ByteArrayOutputStream) generator.getOutputTarget()).toString(StandardCharsets.UTF_8.name());`

Reply:
OK.  I see what you're saying.  The JDK folks have not gotten around to adding the method to accept a discrete `Charset` yet.  Please do use the `.name()` call.